test/odbc/test_pg_datatype_bigint.cpp

```diff
@@ -0,0 +1,361 @@
+#include <gtest/gtest.h>
+#include <sqlext.h>
+#include "odbc_handler.h"
+#include "query_generator.h"
+
+using std::pair;
+
+const string TABLE_NAME = "master_dbo.bigint_table";
+const string COL1_NAME = "pk";
+const string COL2_NAME = "data";
+const string DATATYPE_NAME = "sys.bigint";
+
+vector<pair<string, string>> TABLE_COLUMNS = {
+    {COL1_NAME, DATATYPE_NAME + " PRIMARY KEY"},
+    {COL2_NAME, DATATYPE_NAME}
+  };
+
+class PSQL_DataTypes_Bigint : public testing::Test{
+
+  void SetUp() override {
+  }
+
+  void TearDown() override {
+
+    OdbcHandler test_setup;
+    test_setup.ConnectAndExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+  }
+};
+
+// helper function to convert string to equivalent C version of big int (long long int)
+long long int StringToBigInt(const string &value) {
+  return strtoll(value.c_str(), NULL, 10);
+}
+
+TEST_F(PSQL_DataTypes_Bigint, ColAttributes) {
+
+  const int LENGTH_EXPECTED = 20;
+  const int PRECISION_EXPECTED = 0;
+  const int SCALE_EXPECTED = 0;
+  const string NAME_EXPECTED = "int8";
+  
+  const int BUFFER_SIZE = 256;
+  char name[BUFFER_SIZE];
+  int length;
+  int precision;
+  int scale;
+
+  RETCODE rcode;
+  OdbcHandler odbcHandler;
+
+  // Create a table with columns defined with the specific datatype being tested. 
+  odbcHandler.ConnectAndExecQuery(CreateTableStatement(TABLE_NAME, TABLE_COLUMNS));
+  odbcHandler.CloseStmt();
+
+  // Select * From Table to ensure that it exists
+  odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string> {"pk"}));
```

Comment:
Since COL1_NAME = "pk" is defined, COL1_NAME should be used here instead of "pk"

If you change COL1_NAME definition, this will break.
If specifically primary key is important, COL1_NAME could be named COL1_NAME_PK (but not necessarily)